NetCoreRestApi/DataLayer/Repositories/ProductRepository.cs

```diff
@@ -0,0 +1,98 @@
+﻿using DataLayer.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+
+namespace DataLayer.Repositories
+{
+    public class ProductRepository : IRepository<ProductModel>
+    {
+        private IQueryable<CategoryModel> categoryModels;
+        private readonly Expression<Func<CategoryModel, bool>> IsLaptopsCategory = (x) => x.Name.Equals("Laptops");
+        private readonly Expression<Func<CategoryModel, bool>> IsPritersCategory = (x) => x.Name.Equals("Printers");
+        private readonly Expression<Func<CategoryModel, bool>> IsSaleCategory = (x) => x.Name.Equals("Sale");
```

Comment:
Remove this because  repository doesn't contain any business logic here

Reply:
done